parser.cpp

```diff
@@ -220,14 +232,20 @@ Edl* Parser::parse_import_file()
     if (!t.starts_with("\""))
         ERROR("expecting edl filename");
 
-    Parser p(std::string(t.start_ + 1, t.end_ - 1), searchpaths_);
-    Edl* edl = p.parse();
+    Edl* edl = nullptr;
+    if (pp_.is_included())
+    {
+        Parser p(std::string(t.start_ + 1, t.end_ - 1), searchpaths_, defines_);
+        edl = p.parse();
+    }
     return edl;
 }
 
 void Parser::parse_import()
 {
     Edl* edl = parse_import_file();
+    if (!edl)
+        return;
```

Comment:
Do we want to throw an error and exit it this case?

Reply:
I've added comments for this case.
This is legitimate when the import command is excluded by the directive.  The parser will still proceed without adding anything to the internal data structures. In this case, the edl will be `nullptr`.